typing/parmatch.ml

```diff
@@ -2006,16 +2006,22 @@ let irrefutable pat = le_pat pat omega
 
 let rec inactive pat =
   match pat.pat_desc with
-  | Tpat_lazy _ ->
+  | Tpat_lazy _ | Tpat_array _ ->
       false
-  | Tpat_any | Tpat_var _ | Tpat_constant _ | Tpat_variant (_, None, _) ->
+  | Tpat_any | Tpat_var _ | Tpat_variant (_, None, _) ->
       true
-  | Tpat_tuple ps | Tpat_construct (_, _, ps) | Tpat_array ps ->
+  | Tpat_constant c -> begin
+      match c with
+      | Const_string _ -> false
```

Comment:
We discussed this off-line but I'm putting it here so it's not forgotten: `false` should be changed to `Clflags.unsafe_string`.

Reply:
Rather `not Config.safe_string`.  We need the system to be *configured* with safe strings to be sure that strings cannot be mutated.